src/components/PrimaryPanes/SourcesTree.js

```diff
@@ -116,17 +114,12 @@ class SourcesTree extends Component<Props, State> {
 
   componentWillReceiveProps(nextProps) {
     if (
```

Comment:
we might want to move all of this code into its own util and think through utility

Reply:
Feels good ... SourcesTree seems pretty long with lots of things in it.